### PR TITLE
FindParents methods for groups

### PR DESCRIPTION
**Pull request by savelyev-an**

### CODE REVIEW

File: vault-plugins/flant_iam/model/group.go [hunk 1]
```diff
@@ -255,3 +282,134 @@ func (r *GroupRepository) Sync(objID string, data []byte) error {
...
+				if _, found := resultGroupsSet[candidate]; !found {
+					resultGroupsSet[candidate] = struct{}{}
+					nextSet[candidate] = struct{}{}
+				}
+			}
+		}
+		currentGroupsSet = nextSet
+	}
+	return resultGroupsSet, nil
+}
```
Comment (shvgn) [inline vault-plugins/flant_iam/model/group.go:399]: I see we use string sets a lot. I suggest we add more convenient wrapper for it to reduce struct{}{} boilerplate. Here is an example:

```

// Set is a set of strings. Not concurrency-safe.
type set map[string]struct{}

func (s set) Size() int {
	return len(s)
}

func (s set) Add(el string) {
	s[el] = struct{}{}
}

func newSet(elems ...string) set {
	set := set{}
	for _, el := range elems {
		set.Add(el)
	}
	return set
}

func (s set) Has(el string) bool {
	_, has := s[el]
	return has 
}
```

File: vault-plugins/flant_iam/model/group_test.go [hunk 2]
```diff
@@ -1,12 +1,181 @@
...
+		Groups:          []string{groupUUID2, groupUUID3},
+		ServiceAccounts: []string{serviceAccountUUID2, serviceAccountUUID3},
+		Origin:          OriginIAM,
+	}
+	group5 = Group{
+		UUID:       groupUUID5,
+		TenantUUID: tenantUUID1,
+		Groups:     []string{groupUUID2, groupUUID1},
+		Origin:     OriginIAM,
+	}
```
Comment (shvgn) [inline vault-plugins/flant_iam/model/group_test.go:76]: Just my 2 cents. I'd say that global static fixtures do not scale well. I'd generate them in the scope of a test or in a special helper for it.

File: vault-plugins/flant_iam/model/group_test.go [hunk 3]
```diff
@@ -1,12 +1,181 @@
...
+
+func Test_ListGroups(t *testing.T) {
+	tx := runFixtures(t, tenantFixture, userFixture, serviceAccountFixture, groupFixture).Txn(true)
+	repo := NewGroupRepository(tx)
+
+	ids, err := repo.List(tenantUUID1)
+
+	dieOnErr(t, err)
+	checkDeepEqual(t, []string{groupUUID1, groupUUID2, groupUUID4, groupUUID5}, ids)
+}
```
Comment (shvgn) [inline vault-plugins/flant_iam/model/group_test.go:116]: Setup, action, assertion. LGTM!

**Review by shvgn — APPROVED** [review]
LGTM

File: vault-plugins/flant_iam/model/group.go (hunk 1 above)
Comment (savelyev-an) [reply vault-plugins/flant_iam/model/group.go:399]: We will lost  GroupUUID in signatures in this case...

File: vault-plugins/flant_iam/model/group_test.go (hunk 2 above)
Comment (savelyev-an) [reply vault-plugins/flant_iam/model/group_test.go:76]: this global fixtures are reusable for many tests

**Pull request merged**